### PR TITLE
Add --include-filtered-dependents option

### PR DESCRIPTION
**Pull request by TheNeuralBit**
Repeat of https://github.com/lerna/lerna/pull/1393 for 2.x

## Description

Adds a `localDependents` attribute to `PackageGraphNode`, modifies `addDependencies` to operate more like 3.x and rely on `PackageGraphNode`, and re-uses the BFS for `addDependents`

## Motivation and Context

#611 

## How Has This Been Tested?

Tested the cli on a local repository and added a unit test.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/PackageGraph.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ export class PackageGraphNode {
   constructor(pkg) {
     this.package = pkg;
     this.dependencies = [];
+    this.localDependents = [];
```
Comment (evocateur) [inline src/PackageGraph.js]: Hah, whoops, I forgot 2.x is still using Arrays, and has no concept of `localDependents` or `localDependencies`...

Comment (TheNeuralBit) [reply src/PackageGraph.js]: haha yeah that was a disappointing discovery

**Comment by evocateur** [comment]
As for the local lint failing, 2.x still uses Yarn, so maybe that's the issue? It currently passes for me in my local branch.

In any case, CI likes it, so I'ma merge and publish it. 🎉

**Comment by TheNeuralBit** [comment]
Thanks!

**Comment by evocateur** [comment]
Published in [v2.11.0](https://github.com/lerna/lerna/releases/tag/v2.11.0)

**Pull request merged**